### DISCUSSION TITLE
Chrome download timeout

### DISCUSSION THREAD

**Discussion by cuthulino** [ptarmiganlabs/butler-sheet-icons#346] [Q&A] [answered]
Hi,

ist there a possibility to add chromium manually?

When I execute BSI I get this:
![grafik](https://github.com/ptarmiganlabs/butler-sheet-icons/assets/4634522/beba37d4-99fa-4e0c-b1a2-70e461fa6c5b)

Maybe this is caused by our company webproxy.

I already tried to paste the right chromium Version [1056772](https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win_x64/1056772/) into the right directory `AppData\Local\Temp\.local-chromium\win64-1056772\chrome-win` but it didnt help.

**Comment by cuthulino** [comment, marked as answer]
I managed to get this working.

For everyone else reading this, setting the $env Variables  (HTTP_PROXY and HTTPS_PROXY) including credentials before executing BSI works.

**Reply by mountaindude** [reply]
Great that you got it working.


Do you have any concrete examples how those env variables were set?
If so I can add a sentence or two to the docs on this topic.

Thx

**Reply by mountaindude** [reply]
Something like this I would assume... correct?

`http_proxy=http://USERNAME:PASSWORD@SERVER:PORT/`

**Reply by cuthulino** [reply]
I Think your solution looks right for Linux.

I did it with Windows in Powershell:
```
$proxy = "http://<user>:<password>@<proxyhost>:<proxyport>"
$ENV:HTTP_PROXY=$proxy
$ENV:HTTPS_PROXY=$proxy
```

**Reply by cuthulino** [reply]
I got one more information that could be interesting for the Documentation.
Our System is completely offline and therefore I can not download chromium on the server itself.
You can manually add chromium with these steps (for windows):

1.  Try to run BSI with loglevel "debug" and wait for the failing download
2. Note the chromium version from the Logs (maybe no more needed with new browser parameter)
3. On an other computer with internet connection: Download chromium from Google Data Store: [link](https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win_x64/) (select the right version)
4. Extract the downloaded file so you got a `chrome-win` folder.
5. Move the chrome-win folder to `C:\Users\<UserId>\.cache\puppeteer\chrome\win64-<chromiumVersion>\` (The .exe should be reachable on this path: `C:\Users\<UserId>\.cache\puppeteer\chrome\win64-<chromiumVersion>\chrome-win\chrome.exe`)
